src/apps/miscapps/h3ToGeoBoundaryHier.c

```diff
@@ -94,55 +97,90 @@ void recursiveH3IndexToGeo(H3Index h, int res, int isKmlOut) {
     }
 }
 
-int main(int argc, char* argv[]) {
-    // check command line args
-    if (argc < 2 || argc > 5) {
-        fprintf(stderr, "usage: %s H3Index [resolution outputMode]\n", argv[0]);
-        exit(1);
+int main(int argc, char *argv[]) {
+    int res;
+    H3Index prefixIndex = 0;
+    char userKmlName[BUFF_SIZE] = {0};
+    char userKmlDesc[BUFF_SIZE] = {0};
+
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, if less than the resolution of the prefix "
+                      "only the prefix is printed. Default 0."};
+    Arg prefixArg = {
+        .names = {"-p", "--prefix"},
```

Comment:
Why is this called `prefix` instead of `parent`?

Reply:
Not sure why I did that - I thought perhaps the term `prefix` had been used in these filters before but I see from the diff that isn't the case. I changed it to `parent`.

We might want to refactor these to `h3ToParent` and `h3ToChildren` to match the C API later.

Reply:
Technically all of these app changes could be considered a breaking change requiring a major version bump (if there's anyone out there who has a shell script dependent on them) but I don't think we've considered them part of the API before.

We could make that change if/when we decide to do that (and presumably after we make sure the app collection is capable of doing everything the C API can?)

Reply:
> Technically all of these app changes could be considered a breaking change requiring a major version bump (if there's anyone out there who has a shell script dependent on them) but I don't think we've considered them part of the API before.

We are not considering the filter applications part of the public API at this time.

> We could make that change if/when we decide to do that (and presumably after we make sure the app collection is capable of doing everything the C API can?)

Yes, I mention it to point out an existing terminology which isn't consistent with the C API.